src/runtime/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -20,7 +19,7 @@ interface CrdtCollectionModelMutator {
  * Contains the data that is stored within Pouch
  */
 interface CollectionStorage {
-  model: CrdtCollectionModel;
+  model: CrdtCollectionModel | SerializedModelEntry[];
```

Comment:
When is the SME[] used? Seems like this should have type failed at some point...

Reply:
Yeah this isn't needed. I'm not sure why I added that in :-/

Reply:
Wait no, line 379 has:

doc.model = newModel.toLiteral();

It looks like the model property is used both as a serialized model and a unserialized model. Type checking should definitely have failed for this, but it wasn't because doc is of type PouchDB.Core.ExistingDocument, and we didn't have any type definitions for it, which meant it was assigning it the any type (and skipping typechecking for it). We now have the type definitions, so it's able to type check it properly, and it picked up the error.

Reply:
It looks like doc.model (i.e. the field in CollectionStorage) is always SME[], and never a CrdtCollectionModel - not to be confused with PouchDbCollection._model.

Does removing the CrdtCollectionModel type from this line work?

Reply:
That worked, thanks